script.js

```diff
@@ -1,4 +1,149 @@
+var userWins = 0;
+var compWins = 0;
+var drawCount = 0;
+var firstInput = 0;
+var userName = "";
+var totalGamesPlayed = 0;
+var myOutputValue = "";
+var compFinalChoice = "";
+// below is the random number generator
+var rollDice = function () {
+  // Generate a decimal from 0 through 3, inclusive of 0 and exclusive of 3.
+  var randomDecimal = Math.random() * 3;
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 2 inclusive.
+  var randomInteger = Math.floor(randomDecimal);
+  // Add 1 to get valid dice rolls of 1 through 3 inclusive.
+  var diceNumber = randomInteger + 1;
+  console.log("this is diceNumber " + diceNumber);
+  return diceNumber;
+};
+// below helps to change the random rolls into a number
+var compChoice = function (diceNumber) {
+  if (diceNumber == 1) {
+    compFinalChoice = "scissors";
+    return "scissors";
+  } else if (diceNumber == 2) {
+    compFinalChoice = "paper";
+    return "paper";
+  } else if (diceNumber == 3) {
+    compFinalChoice = "stone";
+    return "stone";
+  }
+};
```

Comment:
Like the use of helper functions for translating number to string
However, comFinalChoice is not necessary  for line 24,27,30
unless you wish to return comFinalChoice between line 32 and 33.